src/compliment/sources/local_bindings.clj

```diff
@@ -38,17 +66,26 @@
             as-bind (conj as-bind)))
 
         (not (#{'& '_} binding-node))
-        [binding-node]))
+        (let [result binding-node
+              candidate (delay (extract-tag-from-bound-to ns bound-to))]
+          (cond-> result
+            (and (not (-> result meta :tag))
+                 @candidate)
+            (vary-meta assoc :tag @candidate)
+
+            true
+            vector))))
```

Comment:
Pretty sure this can be rewritten without `delay` and `cond->`, will be clearer.